EventsExpress.Core/Services/UserService.cs

```diff
@@ -390,5 +395,32 @@ public double GetRating(Guid userId)
                 return 0;
             }
         }
+
+        public IEnumerable<UserDto> GetUsersByNotificationTypes(NotificationChange notificationType, IEnumerable<Guid> userIds)
+        {
+            var usersId = Context.UserNotificationTypes
+                                    .Where(x => x.NotificationTypeId == notificationType && userIds.Contains(x.UserId))
+                                    .Select(x => GetById(x.UserId))
```

Comment:
This `GetById` will be executed in foreach essentially. This is not advised. Just do the mapping to the UserDTO here and that is all.